src/main/java/net/sf/mzmine/modules/peaklistmethods/identification/spectraldbsearch/LocalSpectralDBSearchTask.java

```diff
@@ -110,37 +108,36 @@ public String getTaskDescription() {
   public void run() {
     setStatus(TaskStatus.PROCESSING);
     int count = 0;
-    try {
-      tasks = parseFile(dataBaseFile);
-      totalTasks = tasks.size();
-      if (!tasks.isEmpty()) {
-        // wait for the tasks to finish
-        while (!isCanceled() && !tasks.isEmpty()) {
-          for (int i = 0; i < tasks.size(); i++) {
-            if (tasks.get(i).isFinished() || tasks.get(i).isCanceled()) {
-              count += tasks.get(i).getCount();
-              tasks.remove(i);
-              i--;
-            }
-          }
-          try {
-            Thread.sleep(100);
-          } catch (Exception e) {
-            cancel();
-          }
+    // try {
+    tasks = parseFile(dataBaseFile);
+    totalTasks = tasks.size();
+    // if (!tasks.isEmpty()) {
+    // wait for the tasks to finish
+    while (!isCanceled() && !tasks.isEmpty()) {
+      for (int i = 0; i < tasks.size(); i++) {
+        if (tasks.get(i).isFinished() || tasks.get(i).isCanceled()) {
+          count += tasks.get(i).getCount();
+          tasks.remove(i);
+          i--;
         }
-      } else {
-        setStatus(TaskStatus.ERROR);
-        setErrorMessage("DB file was empty - or error while parsing " + dataBaseFile);
-        throw new MSDKRuntimeException(
-            "DB file was empty - or error while parsing " + dataBaseFile);
       }
-    } catch (Exception e) {
-      logger.log(Level.SEVERE, "Could not read file " + dataBaseFile, e);
-      setStatus(TaskStatus.ERROR);
-      setErrorMessage(e.toString());
-      throw new MSDKRuntimeException(e);
+      try {
+        Thread.sleep(100);
```

Comment:
What is the purpose of this Thread.sleep()?
It looks suspicious. Perhaps add some comments to explain what you're doing.

Reply:
This loop is just waiting for all sub tasks to finish - and then it prints out the total number of library hits and finishes the main task.
Thread.sleep(100); is just to not waste too much time in this thread. 
Any better idea?

Reply:
It’s fine, maybe just add some comments to explain

Reply:
will do